### PR TITLE
[CBRD-21529] Partial page flush

### PR DESCRIPTION
**Pull request by alexstanro**
http://jira.cubrid.org/browse/CBRD-21529

### CODE REVIEW

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/double_write_buffer.h (3 incorrect indentation(s) found)
```
    31 				      DWB_SLOT ** dwb_slot);
    32 extern int dwb_add_page (THREAD_ENTRY * thread_p, FILEIO_PAGE * io_page_p, VPID * vpid, DWB_SLOT * p_dwb_slot);
    36 #endif	/* _DWB_H_ */       /* _DWB_H_ */
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/page_buffer.c (1 incorrect indentation(s) found)
```
  9844   uses_dwb = dwb_is_created (thread_p) && !pgbuf_is_temporary_volume (bufptr->vpid.volid);
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/page_buffer.c (1 incorrect indentation(s) found)
```
  9844   uses_dwb = dwb_is_created (thread_p) && !pgbuf_is_temporary_volume (bufptr->vpid.volid);
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/page_buffer.c (1 incorrect indentation(s) found)
```
  9844   uses_dwb = dwb_is_created (thread_p) && !pgbuf_is_temporary_volume (bufptr->vpid.volid);
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/page_buffer.c (1 incorrect indentation(s) found)
```
  9844   uses_dwb = dwb_is_created (thread_p) && !pgbuf_is_temporary_volume (bufptr->vpid.volid);
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/double_write_buffer.c (1 incorrect indentation(s) found)
```
  3095 
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/double_write_buffer.c (2 incorrect indentation(s) found)
```
   409 STATIC_INLINE int dwb_create_internal (THREAD_ENTRY * thread_p, const char *dwb_volume_name,
  1533 dwb_create_internal (THREAD_ENTRY * thread_p, const char *dwb_volume_name, UINT64 * current_position_with_flags)
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/file_io.c (3 incorrect indentation(s) found)
```
  4307 //      {
  4308 //        ret = dwb_flush_force (thread_p, &all_sync);
  4309 //      }
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/file_io.c (2 incorrect indentation(s) found)
```
  4194 
  4195   fileio_compensate_flush (thread_p, vol_fd, num_pages, skip_flush);
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/double_write_buffer.c (2 incorrect indentation(s) found)
```
  4028 	  iter = 1;
  4050 	      iter++;
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/double_write_buffer.c (10 incorrect indentation(s) found)
```
  1942   DWB_WAIT_QUEUE_ENTRY *queue_entry = (DWB_WAIT_QUEUE_ENTRY *) data;
  1946     {
  1947       woken_thread_p = (THREAD_ENTRY *) queue_entry->data;
  1948       if (woken_thread_p != NULL)
  1949 	{
  1950 	  thread_lock_entry (woken_thread_p);
  1951 	  woken_thread_p->resume_status = THREAD_DWB_QUEUE_RESUMED;
  1952 	  thread_unlock_entry (woken_thread_p);
  1953 	}
  1956 
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/double_write_buffer.c (3 incorrect indentation(s) found)
```
  3901   DWB_BLOCK *initial_block;
  4050 
  4052 #if defined (SERVER_MODE)
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/double_write_buffer.c (13 incorrect indentation(s) found)
```
  4219 		  if (first_partial_flushed_volume == -1)
  4220 		    {
  4221 		      first_partial_flushed_volume = i;
  4222 		    }
  4223 
  4250 	  assert ((first_partial_flushed_volume >= 0)
  4251 		  && ((unsigned int) first_partial_flushed_volume < count_flush_volumes_info));
  4269 	  /* Not all pages written yet, check wether we need to wait for them. */
  4278 	      else
  4279 		{
  4280 		  goto start;
  4281 		}
  4299 	  goto start;
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by eseokoh** [comment]
@alexstanro 

Please synchronize your branch with HEAD.

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/thread/thread.c (118 incorrect indentation(s) found)
```
   296       return tsd_ptr;
   297     }
   395   /* Initialize daemons */
   396   thread_Manager.num_daemons = THREAD_DAEMON_NUM_SINGLE_THREADS;
   397   size = thread_Manager.num_daemons * sizeof (THREAD_DAEMON);
   398   thread_Daemons = (THREAD_DAEMON *) malloc (size);
   399   if (thread_Daemons == NULL)
   400     {
   401       er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, size);
   402       return ER_OUT_OF_VIRTUAL_MEMORY;
   403     }
   404 
   405   /* IMPORTANT NOTE: Daemons are shutdown in the same order as they are created here. */
   406   daemon_index = 0;
   407   shutdown_sequence = 0;
   408 
   409   /* Initialize CSS OOB Handler daemon */
   410   thread_Daemons[daemon_index].type = THREAD_DAEMON_CSS_OOB_HANDLER;
   411   thread_Daemons[daemon_index].daemon_monitor = &thread_Oob_thread;
   412   thread_Daemons[daemon_index].shutdown_sequence = shutdown_sequence++;
```
... More than 20 lines are omitted.
#### File: src/transaction/boot_sr.c (5 incorrect indentation(s) found)
```
  2501   if (error_code != NO_ERROR)
  2502     {
  2503       ASSERT_ERROR ();
  2504       goto error;
  2505     }
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/base/perf_monitor.c (2 incorrect indentation(s) found)
```
   466   PSTAT_METADATA_INIT_COUNTER_TIMER (PSTAT_PB_VICTIM_SEARCH_OWN_PRIVATE_LISTS_CANDIDATE,
   467 				     "alloc_bcb_get_victim_search_own_private_list_assign_direct"),
```
#### File: src/storage/page_buffer.c (5 incorrect indentation(s) found)
```
  8582 		  if (is_own_private_perf_tracking)
  8583 		    {
  8584 		      PERF_UTIME_TRACKER_TIME (thread_p, &perf_tracker,
  8585 					       PSTAT_PB_VICTIM_SEARCH_OWN_PRIVATE_LISTS_ASSIGN_DIRECT);
  8586 		    }
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/page_buffer.c (3 incorrect indentation(s) found)
```
  8237     {
 10133 	  ATOMIC_INC_32 (&pgbuf_Pool.monitor.count_victims_shared_lru_fails, -1);
 10138 	  ATOMIC_INC_32 (&pgbuf_Pool.monitor.count_victims_private_lru_fails, -1);
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/storage/page_buffer.c (1 incorrect indentation(s) found)
```
   825   PGBUF_VICTIM_CANDIDATE_INFO victim_cand_info;
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by eseokoh** [comment]
@alexstanro 

Please synchronize the branch with HEAD. 
It requires to move implementations of daemon threads to new framework.
I see some changes to wrap up the task.

File: src/transaction/log_recovery.c [hunk 1]
```diff
@@ -2443,6 +2443,28 @@ log_recovery_analysis (THREAD_ENTRY * thread_p, LOG_LSA * start_lsa, LOG_LSA * s
 	    }
 	}
 
+      /* Check whether active log pages are corrupted. This may happens in case of partial page flush for instance.  */
+      if ((prm_get_bool_value (PRM_ID_ENABLE_LOG_PAGE_CHECKSUM) == true) && (logpb_is_page_in_archive (log_lsa.pageid)))
```
Comment (eseokoh) [inline src/transaction/log_recovery.c]: @alexstanro 

Please confirm you want `logpb_is_page_in_archive`.

File: src/storage/double_write_buffer.c [hunk 2]
```diff
@@ -0,0 +1,4860 @@
...
+  pthread_mutex_unlock (&slots_hash_entry->mutex);
+  *inserted = 1;
+
+  return NO_ERROR;
+}
+
+/*
+ * dwb_destroy_internal () - Destroy DWB.
+ *
+ * return   : Error co
```
Comment (eseokoh) [inline src/storage/double_write_buffer.c]: Why do we need atomic incr here?

**Pull request merged**